src/runtime/manifest-parser.peg

```diff
@@ -1203,8 +1203,23 @@ SchemaPrimitiveType
     } as AstNode.SchemaPrimitiveType;
   }
 
+SchemaObjectType
```

Comment:
If I understand correctly, this exists so that sanitizeEntry can switch on the kind in order to know whether a string can be converted to an object or not.

I think we want to do this differently:
(1) we should expect that data is converted to/from JSON rather than strings, as that's what has been done elsewhere. This should build directly into Literalizable and avoids parse overhead when we just want a bunch of fields.
(2) we should split the entity classes into a particle-developer facing version and a backend version so that sanitizeEntry doesn't need to cover both deliteralization and sanitization.

Essentially, particle code should never extract entities from JSON (the type-implementing Reference, Instance, etc. classes should be used), and backend code should never construct entities using type-implementing objects. Splitting sanitizeEntry in two will enforce this and also avoid the need to extend field type understanding into the parser, which is an uncomfortable place for that to be.